testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/ResetPasswordTest.java

```diff
@@ -1084,4 +1086,140 @@ public void failResetPasswordServiceAccount() {
         loginPage.assertCurrent();
         assertEquals("Invalid username or password.", errorPage.getError());
     }
+
+    @Test
+    public void resetPasswordLinkNewTabAndProperRedirectAccount() throws IOException {
+        final String REQUIRED_URI = OAuthClient.AUTH_SERVER_ROOT + "/realms/test/account/applications";
+        final String REDIRECT_URI = getAccountRedirectUrl() + "?path=applications";
+        final String CLIENT_ID = "account";
+
+        try (BrowserTabUtil tabUtil = BrowserTabUtil.getInstanceAndSetEnv(driver)) {
+            assertThat(tabUtil.getCountOfTabs(), Matchers.is(1));
+
+            driver.navigate().to(REQUIRED_URI);
+            resetPasswordTwiceInNewTab(defaultUser, CLIENT_ID, false, REDIRECT_URI, REQUIRED_URI);
+            assertThat(driver.getTitle(), Matchers.equalTo("Keycloak Account Management"));
+
+            oauth.openLogout();
+
+            driver.navigate().to(REQUIRED_URI);
+            resetPasswordTwiceInNewTab(defaultUser, CLIENT_ID, true, REDIRECT_URI, REQUIRED_URI);
+            assertThat(driver.getTitle(), Matchers.equalTo("Keycloak Account Management"));
+        }
+    }
+
+    @Test
+    public void resetPasswordLinkNewTabAndProperRedirectAdminConsole() throws IOException {
+        final String REDIRECT_URI = OAuthClient.AUTH_SERVER_ROOT + "/admin/test/console/";
+        final String CLIENT_ID = "security-admin-console";
+
+        // It's necessary to take this here before redirecting to Admin Console.
+        try (BrowserTabUtil tabUtil = BrowserTabUtil.getInstanceAndSetEnv(driver)) {
+            assertThat(tabUtil.getCountOfTabs(), Matchers.is(1));
+            driver.navigate().to(REDIRECT_URI);
+
+            // HtmlUnit has problem with redirecting immediately to Security Console.
+            if (driver instanceof HtmlUnitDriver)
```

Comment:
I suggest to test the scenario with different client than the admin console. Any test with the admin console adds the potential for instabilities and issues and needs of the workarounds like this :) How about the default client "test-client" , which uses the AppPage? For example see the ResetPasswordTest.resetPasswordWithLengthPasswordPolicy() or some other tests used on many places inside the testsuite.

Reply:
I've already found it out too, that a test with admin console requires a lot of effort to execute it properly. The "test-client" will be better choice. Thanks!